### PR TITLE
[datadog_api_key][datadog_application_key] Add exact_match option to data sources

### PR DESCRIPTION
**Pull request by nkzou**
If exact_match is true, searches the returned list of key names for an exact match instead of erroring out due to multiple results.

### CODE REVIEW

**Review by hestonhoffman — APPROVED** [review]

File: datadog/fwprovider/data_source_datadog_application_key.go [hunk 1]
```diff
@@ -85,23 +90,48 @@ func (d *applicationKeyDataSource) Read(ctx context.Context, req datasource.Read
...
+				if state.Name.ValueString() == appKeyAttributes.GetName() {
+					exact_matches++
+					id := appKeyPartialData.GetId()
+					ddResp, _, err := d.Api.GetCurrentUserApplicationKey(d.Auth, id)
+					if err != nil {
+						resp.Diagnostics.Append(utils.FrameworkErrorDiag(err, "error getting application key"))
+						return
+					}
+					appKeyData := ddResp.GetData()
+					d.updateState(&state, &appKeyData)
```
Comment (HantingZhang2) [inline datadog/fwprovider/data_source_datadog_application_key.go:114]: Is it necessary to update the state no matter what the exact_match count is ?

File: datadog/fwprovider/data_source_datadog_api_key.go [hunk 2]
```diff
@@ -89,25 +93,48 @@ func (d *apiKeyDataSource) Read(ctx context.Context, req datasource.ReadRequest,
...
+				if state.Name.ValueString() == apiKeyAttributes.GetName() {
+					exact_matches++
+					id := apiKeyPartialData.GetId()
+					ddResp, _, err := d.Api.GetAPIKey(d.Auth, id)
+					if err != nil {
+						resp.Diagnostics.Append(utils.FrameworkErrorDiag(err, "error getting api key"))
+						return
+					}
+					apiKeyData := ddResp.GetData()
+					d.updateState(&state, &apiKeyData)
```
Comment (HantingZhang2) [inline datadog/fwprovider/data_source_datadog_api_key.go:117]: Same here

File: datadog/fwprovider/data_source_datadog_application_key.go (hunk 1 above)
Comment (nkzou) [reply datadog/fwprovider/data_source_datadog_application_key.go:114]: good catch, will update things to only updateState once at the end

**Review by HantingZhang2 — APPROVED** [review]

**Comment by arunsisodiya** [comment]
Hey @nkzou ,

Do we know when this change will be available in the provider (in which version)?
I am waiting for this due to my use-cases related to Datadog API keys and application keys.

**Comment by nkzou** [comment]
@arunsisodiya https://github.com/DataDog/terraform-provider-datadog/releases/tag/v3.36.0 is released now, please try it out, thanks!

**Pull request merged**